src/local-introspection.js

```diff
@@ -35,35 +36,44 @@ module.exports = (options) => {
 
   return async function localIntrospect(token, tokenTypeHint) {
     if (!jwksFetchKey) {
-      throw new Error('Neither `jwks` or `jwks_uri` defined');
+      throw new errors.ConfigurationError('Neither `jwks` or `jwks_uri` defined');
     }
 
     if (tokenTypeHint !== 'access_token') {
       debug('Not an access token, tokenTypeHint=%s', tokenTypeHint);
-      throw new Error('Only access tokens are supported for local introspection');
+      throw new errors.MalformedTokenError('Only access tokens are supported for local introspection');
     }
 
     const decodedToken = jwt.decode(token, { complete: true });
     if (!decodedToken) {
       debug('Not a JWT token');
-      throw new Error('Token is not a JWT');
+      throw new errors.MalformedTokenError('Token is not a JWT');
     }
 
     if (!decodedToken.header.kid) {
       debug('Tokens does not contain kid in header');
-      throw new Error('Token does not contain kid in header');
+      throw new errors.MalformedTokenError('Token does not contain kid in header');
     }
 
     let pem;
     try {
       const key = await jwksFetchKey(decodedToken.header.kid);
       pem = key.publicKey || key.rsaPublicKey;
     } catch (err) {
-      throw new Error('Could not find key matching kid');
+      throw new errors.MalformedTokenError('Could not find key matching kid');
     }
 
-    // jwtVerify may throw, but index will catch
-    const verified = await jwtVerify(token, pem, { algorithms: options.allowed_algs });
-    return Object.assign({ active: true }, verified);
+    try {
+      const verified = await jwtVerify(token, pem, { algorithms: options.allowed_algs });
+      return Object.assign({ active: true }, verified);
+    } catch (err) {
+      if (err instanceof jwt.TokenExpiredError) {
+        throw new errors.TokenExpiredError();
+      }
+      if (err instanceof jwt.NotBeforeError) {
+        throw new errors.NotBeforeError();
+      }
+      throw new errors.IntrospectionError();
```

Comment:
Should this preserve the message from `err`?
Otherwise it doesn't give much information.